cmd/service-controller/console_server.go

```diff
@@ -354,24 +385,36 @@ func (server *ConsoleServer) listen() {
 		addr = os.Getenv("METRICS_HOST") + addr
 	}
 	log.Printf("Console server listening on %s", addr)
-	http.Handle("/DATA", authenticated(server))
-	http.Handle("/version", authenticated(server.version()))
-	http.Handle("/events", authenticated(server.serveEvents()))
-	http.Handle("/servicecheck/", server.checkService())
-	http.Handle("/", authenticated(http.FileServer(http.Dir("/app/console/"))))
-	log.Fatal(http.ListenAndServe(addr, nil))
+	r := mux.NewRouter()
+	r.Handle("/DATA", authenticated(server))
+	r.Handle("/tokens", authenticated(serveTokens(server.tokens)))
+	r.Handle("/tokens/", authenticated(serveTokens(server.tokens)))
+	r.Handle("/tokens/{name}", authenticated(serveTokens(server.tokens)))
+	r.Handle("/downloadclaim/{name}", authenticated(downloadClaim(server.tokens)))
+	r.Handle("/links", authenticated(serveLinks(server.links)))
+	r.Handle("/links/", authenticated(serveLinks(server.links)))
+	r.Handle("/links/{name}", authenticated(serveLinks(server.links)))
+	r.Handle("/version", authenticated(server.version()))
+	r.Handle("/site", authenticated(server.site()))
+	r.Handle("/events", authenticated(server.serveEvents()))
+	r.Handle("/servicecheck/{name}", server.checkService())
+	r.PathPrefix("/").Handler(authenticated(http.FileServer(http.Dir("/app/console/"))))
+	if os.Getenv("USE_CORS") != "" {
+		r.Use(cors)
+	}
+	log.Fatal(http.ListenAndServe(addr, r))
 }
 
 func (server *ConsoleServer) listenLocal() {
 	addr := "localhost:8181"
-	mux := http.NewServeMux()
-	mux.Handle("/DATA", server)
-	mux.Handle("/version", server.version())
-	mux.Handle("/events", server.serveEvents())
-	mux.Handle("/sites", server.serveSites())
-	mux.Handle("/services", server.serveServices())
-	mux.Handle("/servicecheck/", server.checkService())
-	log.Fatal(http.ListenAndServe(addr, mux))
+	r := mux.NewRouter()
+	r.Handle("/DATA", server)
+	r.Handle("/version", server.version())
+	r.Handle("/events", server.serveEvents())
+	r.Handle("/sites", server.serveSites())
+	r.Handle("/services", server.serveServices())
+	r.Handle("/servicecheck/", server.checkService())
```

Comment:
@grs guess you need the same url change here or the `get` command will fail.

Reply:
Sorry, yes! Done.